### PR TITLE
Migrate from service_detail to info_detail

### PR DESCRIPTION
**Pull request by weitzman**
**Jira:** (Skip unless you are MA staff)
DP-26154

**To Test:**
- [  ] Using Prod as a reference, compare the view and edit tabs for a few service detail pages.

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: scripts/migrate_service_details.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# drush sset entity_hierarchy_disable_writes 1
+
+#Show migration status. Re-run anytime to see the current status.
+drush migrate:status
+
+#This is the long migration that creates info_details nodes.
+drush migrate:import service_details
+
+#This updates entity refs and link fields and other usages of the old service_detail nodes
+drush migrate:import update_references
```
Comment (tom-fleming) [inline scripts/migrate_service_details.sh:14]: update_references_nodes

Comment (tom-fleming) [inline scripts/migrate_service_details.sh]:
```suggestion
drush migrate:import update_references_node

#This updates entity refs and link fields and other usages of the old service_detail paragraphs
drush migrate:import update_references_paragraph
```

File: docroot/modules/custom/mass_migrate/src/Commands/MassMigrateCommands.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    \Drupal::service('config.factory')->clearStaticCache();
+    \Drupal::service('cache.discovery')->deleteAll();
+  }
+
+  /**
+   * Override config after running drush migrate:import.
+   *
+   * @hook post-command migrate:import
+   */
+  public function overrideMigrateImportPostCommand() {
```
Comment (tom-fleming) [inline docroot/modules/custom/mass_migrate/src/Commands/MassMigrateCommands.php:30]:
```suggestion
  public function overrideMigrateImportPostCommand() {
    // Usage updates now gfo into a queue which we will process at end of migration.
    $GLOBALS['config']['entity_usage.settings']['track_enabled_source_entity_types'] = ['placeholder', 'another'];
    
```

File: docroot/sites/default/settings.acquia.php [hunk 3]
```diff
@@ -82,7 +82,8 @@
  *
  * @see https://docs.acquia.com/articles/password-protect-your-non-production-environments-acquia-hosting#phpfpm
  */
-if (!$cli && !$is_prod && !in_array($_SERVER['SERVER_NAME'], ['wwwcf.digital.mass.gov', 'editcf.digital.mass.gov', 'stage.mass.gov', 'edit.stage.mass.gov'])) {
+// @todo remove feature2
```
Comment (tom-fleming) [inline docroot/sites/default/settings.acquia.php:85]: Marking as todo

File: conf/drupal/config/field.storage.node.field_migrated_node_id.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+uuid: ccad061d-64c4-4c4c-ad6a-074c1f25f28b
+langcode: en
+status: true
+dependencies:
+  module:
+    - node
+id: node.field_migrated_node_id
+field_name: field_migrated_node_id
+entity_type: node
+type: string
```
Comment (weitzman) [inline conf/drupal/config/field.storage.node.field_migrated_node_id.yml]: Maybe it doesnt matter, but I'm surprised tis isnt an integer field

File: docroot/modules/custom/mass_migrate/mass_migrate.module [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * Implements hook_module_implements_alter().
+ */
+function mass_migrate_module_implements_alter(&$implementations, $hook) {
+  // Completely take over entity_hierarchy delete hook.
+  if ($hook == 'entity_delete' && \Drupal::state()->get('entity_hierarchy_disable_writes', FALSE)) {
+    unset($implementations['entity_hierarchy']);
+  }
+}
```
Comment (weitzman) [inline docroot/modules/custom/mass_migrate/mass_migrate.module:14]: FYI we have Hux installed and it has this feature more directly available - https://www.drupal.org/project/hux

File: docroot/modules/custom/mass_migrate/src/Commands/MassMigrateCommands.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  /**
+   * Override config when running drush entity:delete.
+   *
+   * @hook post-command entity:delete
+   */
+  public function overrideEntityDeletePostCommand() {
+    // Unset variables after running the command.
+    \Drupal::state()->set('entity_hierarchy_disable_writes', FALSE);
+  }
```
Comment (weitzman) [inline docroot/modules/custom/mass_migrate/src/Commands/MassMigrateCommands.php]: This seems like an awkward way to set `entity_hierarchy_disable_writes`. If we leave mass_migrate enabled, we will be enabling and disabling whenever the entity:delete drush command runs. Thats true even after the migration is complete. Could we add these Drush commands to the script instead?

File: docroot/modules/custom/mass_migrate/src/Commands/MassMigrateCommands.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    \Drupal::service('cache.discovery')->deleteAll();
+  }
+
+  /**
+   * Override config after running drush migrate:import.
+   *
+   * @hook post-command migrate:import
+   */
+  public function overrideMigrateImportPostCommand() {
+    // Usage updates now gfo into a queue which we will process at end of migration.
```
Comment (weitzman) [inline docroot/modules/custom/mass_migrate/src/Commands/MassMigrateCommands.php]:
```suggestion
    // Usage updates now go into a queue which we will process at end of migration.
```

**Review by weitzman — COMMENTED** [review]
LGTM. A few minor comments below.

File: conf/drupal/config/field.storage.node.field_migrated_node_id.yml (hunk 4 above)
Comment (tom-fleming) [reply conf/drupal/config/field.storage.node.field_migrated_node_id.yml]: Yeah, agreed. Changed this to an integer field.

File: docroot/modules/custom/mass_migrate/src/Commands/MassMigrateCommands.php (hunk 6 above)
Comment (tom-fleming) [reply docroot/modules/custom/mass_migrate/src/Commands/MassMigrateCommands.php]: I moved this over to the drush command script.

File: docroot/modules/custom/mass_migrate/mass_migrate.module (hunk 5 above)
Comment (tom-fleming) [reply docroot/modules/custom/mass_migrate/mass_migrate.module:14]: I left this one in place, but also good to know. Mentioning @arthurbaghdas here just so he sees this.

**Pull request merged**